homeassistant/components/spc/__init__.py

```diff
@@ -40,41 +43,58 @@
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the SPC component."""
 
-    async def async_upate_callback(spc_object):
-        if isinstance(spc_object, Area):
-            async_dispatcher_send(hass, SIGNAL_UPDATE_ALARM.format(spc_object.id))
-        elif isinstance(spc_object, Zone):
-            async_dispatcher_send(hass, SIGNAL_UPDATE_SENSOR.format(spc_object.id))
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+
+    return True
+
+
+async def _async_upate_callback(hass, spc_object):
+    if isinstance(spc_object, Area):
+        async_dispatcher_send(hass, SIGNAL_UPDATE_ALARM.format(spc_object.id))
+    elif isinstance(spc_object, Zone):
+        async_dispatcher_send(hass, SIGNAL_UPDATE_SENSOR.format(spc_object.id))
 
-    session = aiohttp_client.async_get_clientsession(hass)
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SPC from a config entry."""
+    config = entry.data
 
     spc = SpcWebGateway(
         loop=hass.loop,
-        session=session,
-        api_url=config[DOMAIN].get(CONF_API_URL),
-        ws_url=config[DOMAIN].get(CONF_WS_URL),
-        async_callback=async_upate_callback,
+        session=async_create_clientsession(hass),
+        api_url=config.get(CONF_API_URL),
+        ws_url=config.get(CONF_WS_URL),
+        async_callback=lambda spc_object: _async_upate_callback(hass, spc_object),
     )
 
     hass.data[DATA_API] = spc
```

Comment:
Looks like this would be a problem if you setup multiple entries of spc

Reply:
The `spc_object.id` would be different so should be OK.

Reply:
Changed it to make use of the entry.entity_id